### PR TITLE
New thread for ebpf.plugin

### LINKED ISSUE #10725: Sync monitoring
If the syscall  `sync` is called frequently, this can create performance bottleneck, netdata needs to bring a monitor for this using eBPF plugin.  

This plugin will have one chart demonstrating the frequency that the syscall is called and we will also have an alarm for it.

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fixes #10725

This PR is bringing the first step to monitor synchronization syscalls.
##### Component Name
ebpf.plugin
##### Test Plan

1 - Compile this PR.
2 - Run the following commands to create a peak on chart:

```bash
sync; echo 1 > /proc/sys/vm/drop_caches
```

##### Additional Information

This PR was tested on 

| Linux Distribution | Kernel Version |
|-------------------------|---------------------|
| Slackware Current | 5.11.2 |
| Slackware Current | 5.10.2[012] |
| Slackware Current | 5.11.2 |
| Slackware Current | 5.4.95 |
| Slackware Current | 4.16.18 |
| Ubuntu 18.04 | 5.4.66 |
| Ubuntu 18.04 | 4.18.0-23 |
| Ubuntu 18.04 | 4.15.0 |
| CentOS 7.9 | 3.10.0 |

### CODE REVIEW

**Comment by Ferroin** [comment]
Overall, we would want to monitor the following syscalls for this:

* msync
* fsync
* fdatasync
* sync
* sync_file_range
* syncfs

We probably also want an option in the collector itself to choose to split them out as separate dimensions or have them aggregated into a single dimension. The aggregate value is useful for alerting, but when figuring out what is causing any issues, you almost certainly want the individual counts.

**Comment by thiagoftsm** [comment]
> Overall, we would want to monitor the following syscalls for this:
> 
>     * msync
> 
>     * fsync
> 
>     * fdatasync
> 
>     * sync
> 
>     * sync_file_range
> 
>     * syncfs
> 
> 
> We probably also want an option in the collector itself to choose to split them out as separate dimensions or have them aggregated into a single dimension. The aggregate value is useful for alerting, but when figuring out what is causing any issues, you almost certainly want the individual counts.

Thank you @Ferroin , this PR was based in [syncsnoop](https://github.com/iovisor/bcc/blob/master/tools/syncsnoop.py) that monitors only one syscall, but I agree with you that it will be more useful to bring all these `syscalls`. I will move this PR forward with only this `syscall` and after it is merged,  I will make another bringing the other `syscalls`.

**Review by underhood — APPROVED** [review]

File: collectors/ebpf.plugin/ebpf.c [hunk 1]
```diff
@@ -1783,6 +1815,7 @@ static void parse_args(int argc, char **argv)
         {"net",       no_argument,    0,  'n' },
         {"process",   no_argument,    0,  'p' },
         {"return",    no_argument,    0,  'r' },
+        {"sync", no_argument,    0,  's' },
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf.c:1818]:
```suggestion
        {"sync",      no_argument,    0,  's' },
```

File: collectors/ebpf.plugin/ebpf_sync.c [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+#include "ebpf.h"
+#include "ebpf_sync.h"
+
+static ebpf_data_t sync_data;
+
+static struct bpf_link **probe_links = NULL;
+static struct bpf_object *objects = NULL;
+
+
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_sync.c:11]:
```suggestion
```

File: collectors/ebpf.plugin/ebpf_sync.c [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+static netdata_publish_syscall_t sync_counter_publish_aggregated;
+
+static int read_thread_closed = 1;
+
+static int *map_fd = NULL;
+static netdata_idx_t sync_hash_values = 0;
+
+struct netdata_static_thread sync_threads = {"SYNC KERNEL",
+                                                  NULL, NULL, 1, NULL,
+                                                  NULL,  NULL};
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_sync.c:23]: Please fix the formatting.

File: collectors/ebpf.plugin/ebpf_sync.c [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+/*****************************************************************
+ *
+ *  DATA THREAD
+ *
+ *****************************************************************/
+
+/**
+ * Read global counter
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_sync.c:32]: `counter` -> `table`?

**Review by vlvkobal — COMMENTED** [review]
Except for some minor formatting issues LGTM.

**Review by vlvkobal — APPROVED** [review]

**Review by underhood — APPROVED** [review]

**Pull request merged**